### DISCUSSION TITLE
Use Generated Class in query.graphql

### DISCUSSION THREAD

**Discussion by andresmonc** [Netflix/dgs-framework#1265] [Q&A]
Hello - I'm trying to use a class that's imported from a dependency in my query.graphql schema.

I've added the type mapping in my gradle file
```
generateJava{
	schemaPaths = ["${projectDir}/src/main/resources/graphql", "${projectDir}/src/main/resources/*"] // List of directories containing schema files
	packageName = 'com.xyz.gen' // The package name to use to generate sources
	generateClient = true // Enable generating the type safe query API
	typeMapping = ["Session": "com.xyz.shared.xyz.xas.Session"] // Classes used during code generation.
}

```

Is there anything else I need to do? I keep getting this error
	 * The field type 'Session' is not present when resolving type 'Query' [@1:1]

**Comment by srinivasankavitha** [comment]
Does your schema actually have a `Session` type defined?

**Comment by idiazvalenzuela** [comment]
I have a similar issue. 
My schema has a Currency enum that the plugin wrongly resolves as java.util.Currency. 
If I add my Currency type to typeMapping, then my custom Currency.kt it's not generated.

Can you help me with this? I'm using kotlin and latest version 5.6.2

Thanks

**Comment by srinivasankavitha** [comment]
You are running into this with Currency because codegen implicitly maps it
to the known Currency class. This can be overridden using the config for
type mapping, but you can't use a class that needs to be generated by the
same codegen plugin. I think we can remove the mapping in codegen here for
both java and kotlin, and that should be fine:
https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/TypeUtils.kt#L44

Feel free to open a PR if this is blocking you, else we will fix it in the
next release this week.

**Comment by idiazvalenzuela** [comment]
Thanks so much for your help, here is the PR: https://github.com/Netflix/dgs-codegen/pull/488

**Comment by idiazvalenzuela** [comment]
Thanks again for your help. 
Is it possible to get a released version that includes this? We are currently blocked, so we will be really grateful if there's a release that includes this.

**Comment by srinivasankavitha** [comment]
I'm planning to release today.

**Comment by srinivasankavitha** [comment]
The latest release `v5.6.3` is now available
